plugins/flytekit-papermill/flytekitplugins/papermill/task.py

```diff
@@ -196,21 +198,17 @@ def rendered_output_path(self) -> str:
         return self._notebook_path.split(".ipynb")[0] + "-out.html"
 
     def get_container(self, settings: SerializationSettings) -> task_models.Container:
-        # The task name in the original command is incorrect because we use _dummy_task_func to construct the _config_task_instance.
-        # Therefore, Here we replace the original command with NotebookTask's command.
-        def fn(settings: SerializationSettings) -> typing.List[str]:
-            return self.get_command(settings)
-
-        self._config_task_instance.set_command_fn(fn)
+        # Always extract the module from the notebook task, no matter what _config_task_instance is.
+        _, m, t, _ = extract_task_module(self)
+        loader_args = ["task-module", m, "task-name", t]
+        self._config_task_instance.task_resolver.loader_args = lambda ss, task: loader_args
         return self._config_task_instance.get_container(settings)
 
     def get_k8s_pod(self, settings: SerializationSettings) -> task_models.K8sPod:
-        # The task name in original command is incorrect because we use _dummy_task_func to construct the _config_task_instance.
-        # Therefore, Here we replace primary container's command with NotebookTask's command.
-        def fn(settings: SerializationSettings) -> typing.List[str]:
-            return self.get_command(settings)
-
-        self._config_task_instance.set_command_fn(fn)
+        # Always extract the module from the notebook task, no matter what _config_task_instance is.
+        _, m, t, _ = extract_task_module(self)
+        loader_args = ["task-module", m, "task-name", t]
+        self._config_task_instance.task_resolver.loader_args = lambda ss, task: loader_args
```

Comment:
This is overriding the task resolver `loader_args` function, which is a problem in the case of multiple tasks sharing the same task resolver (i.e. the [default_task_resolver](https://github.com/flyteorg/flytekit/blob/88178a7b86848807c31a7fcb43c161bbc9e003ce/flytekit/core/python_auto_container.py#L273)).